aws/resource_aws_instance.go

```diff
@@ -783,6 +800,13 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 			}
 		}
 	}
+	{
+		creditSpecifications, err := getCreditSpecifications(conn, d.Id())
+		if err != nil {
+			return err
+		}
+		d.Set("credit_specification", creditSpecifications)
```

Comment:
Same here -- we should return errors when attempting to set non-scalar attributes in the state (otherwise it will silently use the value of the configuration or in the case of the data source, not set it at all):

```go
if err := d.Set("credit_specification", creditSpecifications); err != nil {
  return fmt.Errorf("error setting credit_specification: %s", err)
}
```